controllers/admin/admin_media_controller.py

```diff
@@ -26,6 +26,21 @@ def get(self):
         self.response.out.write(template.render(path, self.template_values))
 
 
+class AdminMediaDeleteReference(LoggedInHandler):
+    def post(self, media_key_name):
+        self._require_admin()
+
+        media = Media.get_by_id(media_key_name)
+
+        media.references.remove(media.create_reference(
+            self.request.get("reference_type"),
+            self.request.get("reference_key_name")))
+
+        media.put() ## Todo, update manipulators to support removing an item that ends a list
```

Comment:
I'm running into a weird problem.

When I delete the reference here, then go to do a `MediaManipulator.createOrUpdate()`, the "old model" the MediaManipulator is getting back from its internal ndb call already has `references = []`, even though the only place that mutation has happened is against the media object I am attempting to write back to the database. It's like the "uncommitted changes" are leaking back in to the DB.

I can tell this is happening because if I comment out the reference removal, the objects come back with references.

### Comment out removal
```
old_models = ndb.get_multi([ndb.Key(type(model).__name__, model.key_name) for model in new_models])
print "ze old models %s" % old_models

>> ze old models [Media(key=Key('Media', 'youtube_dQw4w9WgXcQ'), created=datetime.datetime(2016, 2, 27, 22, 50, 10, 257700), details_json=None, foreign_key=u'dQw4w9WgXcQ', media_type_enum=0, private_details_json=None, references=[Key('Team', 'frc1')], updated=datetime.datetime(2016, 3, 3, 5, 1, 21, 456624), year=2016)]
```

### Leave Removal In
```
old_models = ndb.get_multi([ndb.Key(type(model).__name__, model.key_name) for model in new_models])
print "ze old models %s" % old_models

ze old models [Media(key=Key('Media', 'youtube_dQw4w9WgXcQ'), created=datetime.datetime(2016, 2, 27, 22, 50, 10, 257700), details_json=None, foreign_key=u'dQw4w9WgXcQ', media_type_enum=0, private_details_json=None, references=[], updated=datetime.datetime(2016, 3, 3, 5, 1, 21, 456624), year=2016)]
```

Either way, the Manipulator decides the object isn't dirty and doesn't bother putting it back in the database. If I manually add `old_media.dirty = True` to the bottom of the `MediaManipulator.updateMerge()` method, it does successfully wipe itself.

Why am I pulling the already-modified media out the ndb with that query? :/ Is this a dev server only problem, or will it exist in prod?